### PR TITLE
feat: Add dllm generation support

### PR DESCRIPTION
**Pull request by pthombre**
What does this PR do?
                                                                                                                                                                                                                                                                                                                                                    
  Add an inference/generation script for Diffusion LLM (LLaDA) checkpoints fine-tuned via Automodel.
                                                                                                                                                                                                                                                                                                                                                    
  Changelog
                                                                                                                                                                                                                                                                                                                                                    
  - Add examples/dllm_generate/generate.py — a self-contained script for running inference on Automodel LLaDA checkpoints using MDLM (Masked Diffusion Language Model) sampling.                                                                                                                                                                    
  - Implement flexible checkpoint path resolution that accepts run dirs, step dirs, or consolidated HF-format dirs.
  - Support three generation modes:                                                                                                                                                                                                                                                                                                                 
    - Chat generation — wraps prompts with the LLaDA chat template and generates responses.
    - Raw generation (--raw) — continues text without a chat template.                                                                                                                                                                                                                                                                              
    - Infilling (--infill) — replaces [MASK] placeholders in the prompt via in-place diffusion unmasking.                                                                                                                                                                                                                                           
  - Implement block-wise iterative MDLM unmasking with configurable diffusion steps, block size, temperature, and remasking strategy (low_confidence or random).                                                                                                                                                                                    
  - Support batched generation across multiple --prompt arguments.                                                                                                                                                                                                                                                                                  
  - Use NeMoAutoModelForCausalLM and NeMoAutoTokenizer from nemo_automodel for model/tokenizer loading.                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                                                                                    
  Before your PR is "Ready for review"
                                                                                                                                                                                                                                                                                                                                                    
  Pre checks:                                                                                                                                                                                                                                                                                                                                       
  
  - Make sure you read and followed Contributor guidelines                                                                                                                                                                                                                                                                                          
  - Did you write any new necessary tests?
  - Did you add or update any necessary documentation?                                                                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                                                                                    
  If you haven't finished some of the above items you can still open "Draft" PR.                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                                                    
  Additional Information                                                                                                                                                                                                                                                                                                                            
                  
  How to run

  # Basic chat generation
  python examples/dllm_generate/generate.py \                                                                                                                                                                                                                                                                                                       
      --checkpoint checkpoints/llada-tulu3-500/LATEST/model/consolidated \
      --prompt "Explain what a neural network is."                                                                                                                                                                                                                                                                                                  
                  
  # Multiple prompts                                                                                                                                                                                                                                                                                                                                
  python examples/dllm_generate/generate.py \
      --checkpoint checkpoints/llada-tulu3-500/LATEST/model/consolidated \                                                                                                                                                                                                                                                                          
      --prompt "How far does Lily run in 8 hours?" \
      --prompt "Write an educational python function."                                                                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                                                                                    
  # Custom sampling parameters                                                                                                                                                                                                                                                                                                                      
  python examples/dllm_generate/generate.py \                                                                                                                                                                                                                                                                                                       
      --checkpoint checkpoints/llada-tulu3-1k/LATEST/model/consolidated \
      --prompt "Write a haiku about programming." \                                                                                                                                                                                                                                                                                                 
      --steps 256 --max_new_tokens 256 --temperature 0.2                                                                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                                                                                                    
  # Raw mode (no chat template)                                                                                                                                                                                                                                                                                                                     
  python examples/dllm_generate/generate.py \
      --checkpoint checkpoints/llada-tulu3-500/LATEST/model/consolidated \                                                                                                                                                                                                                                                                          
      --prompt "Once upon a time" --raw
                                                                                                                                                                                                                                                                                                                                                    
  # Infilling mode
  python examples/dllm_generate/generate.py \
      --checkpoint checkpoints/llada-tulu3-500/LATEST/model/consolidated \                                                                                                                                                                                                                                                                          
      --prompt "The capital of France is [MASK] and it is known for [MASK]." \
      --infill

### CODE REVIEW

**Comment by akoumpa** [comment]
ci passed https://github.com/NVIDIA-NeMo/Automodel/actions/runs/24044433659/job/70124126521?pr=1692
